docker/datahub-ingestion/Dockerfile

```diff
@@ -140,7 +140,9 @@ RUN --mount=type=bind,source=./python-build/version_updater.py,target=/version_u
 
 FROM add-code AS install-slim
 
+# zstd needs to be pinned because the latest version causes issues on arm
 RUN --mount=type=cache,target=$HOME/.cache/uv,uid=1000,gid=1000 \
+    UV_LINK_MODE=copy uv pip install "zstd<1.5.6.8" && \
```

Comment:
This is here because we don't want to add a restriction to metadata-ingestion? I think we need to think about locking all versions for the image builds. Is there a path to generating a requirements.txt for images specifically with  pinned versions for all packages?